pipeline/trusted-resources/cmd/sign/main.go

```diff
@@ -71,36 +73,49 @@ func main() {
 	}
 	defer f.Close()
 
-	tsBuf, err := ioutil.ReadFile(*taskFile)
+	tsBuf, err := ioutil.ReadFile(*resourceFile)
 	if err != nil {
-		log.Fatalf("error reading task: %v", err)
+		log.Fatalf("error reading file: %v", err)
 	}
 
-	ts := &v1beta1.Task{}
-	if err := yaml.Unmarshal(tsBuf, &ts); err != nil {
-		log.Fatalf("error unmarshalling taskrun: %v", err)
+	var crd resourcesemantics.GenericCRD
+	switch *kind {
+	case "Task":
+		crd = &v1beta1.Task{}
+	case "Pipeline":
+		crd = &v1beta1.Pipeline{}
+	}
+	if err := yaml.Unmarshal(tsBuf, &crd); err != nil {
+		log.Fatalf("error unmarshalling Task/Pipeline: %v", err)
 	}
-
 	// Sign the task and write to writer
-	if err := SignTask(ctx, ts, signer, f); err != nil {
-		log.Fatalf("error signing taskrun: %v", err)
+	if err := Sign(ctx, crd, signer, f); err != nil {
+		log.Fatalf("error signing Task/Pipeline: %v", err)
 	}
 
 }
 
-// Sign the taskrun and output signed bytes to writer
-func SignTask(ctx context.Context, task *v1beta1.Task, signer sigstore.Signer, writer io.Writer) error {
-	sig, err := trustedtask.SignInterface(signer, task)
+// Sign the crd and output signed bytes to writer
+func Sign(ctx context.Context, crd resourcesemantics.GenericCRD, signer sigstore.Signer, writer io.Writer) error {
+	sig, err := trustedtask.SignInterface(signer, crd)
 	if err != nil {
 		return err
 	}
 
-	if task.Annotations == nil {
-		task.Annotations = map[string]string{}
+	switch crd.GetObjectKind().GroupVersionKind().Kind {
+	case "Task":
+		if crd.(*v1beta1.Task).Annotations == nil {
+			crd.(*v1beta1.Task).Annotations = map[string]string{}
+		}
+		crd.(*v1beta1.Task).Annotations[trustedtask.SignatureAnnotation] = sig
+	case "Pipeline":
+		if crd.(*v1beta1.Pipeline).Annotations == nil {
+			crd.(*v1beta1.Pipeline).Annotations = map[string]string{}
+		}
+		crd.(*v1beta1.Pipeline).Annotations[trustedtask.SignatureAnnotation] = sig
```

Comment:
Instead of using `GenericCRD`, consider using [`metav1.Object`](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Object) instead. And use the Get/SetAnnotations methods.

Then this looks like this for *any* type!

```suggestion
	a := o.GetAnnotations()
	a[trustedtask.SignatureAnnotation] = sig
	o.SetAnnotations(a)
```

Reply:
yeah this works! Thanks!